### DISCUSSION TITLE
After migrating to PHP 8.1 and 5.0.0: Hashes don't match anymore

### DISCUSSION THREAD

**Discussion by Nyde** [phalcon/cphalcon#16117] [Q&A] [answered]
Good morning,

currently, I'm migrating from PHP 7.4 with Phalcon 4.1.0 to PHP 8.1.6 with Phalcon 5. On the new version, my hashes (from Phalcon\Crypt, now Phalcon\Encryption\Crypt) don't work anymore, with the key remaining the same. When I try to decrypt the hashes, I just get ```Hash does not match.```.

The code in question:

```php
$di->set(
    'crypt',
    function () {
        $crypt = new Crypt();
        $crypt->setKey(
            "32 characters long key here"
        );

        return $crypt;
    },
    true
);
```
I didn't change any code related to this, except the new namespace.

I found this issue #16010 with someone having the same problem and I wondered if this has been further investigated after this. I really want to avoid rehashing all values in store...

Thank you.

**Comment by s-ohnishi** [comment, marked as answer]
I have not been able to understand what processing is actually done internally, but when I look at the published source, it seems that the default value of the argument `useSigning` is different.

Phalcon\Crypt v4.1.x
```
public function __construct(string! cipher = "aes-256-cfb", bool useSigning = false)
```
Phalcon\Encryption\Crypt v5.0.x (master)
```
public function __construct(
         string cipher = self::DEFAULT_CIPHER,
         bool useSigning = true,
         <PadFactory> padFactory = null
     ) {
```
Can't you solve it by calling this with the same value?

**Reply by Nyde** [reply]
Yeah, that's it. I disabled signing and the old key works again. Thank you for this hint.
